openmdao/utils/cs_safe.py

```diff
@@ -41,3 +41,31 @@ def norm(x, axis=None):
     ndarray
     """
     return np.sqrt(np.sum(x**2, axis=axis))
+
+
+def arctan2(y, x):
+    """
+    Numpy-compatible, complex-compatible arctan2 function for use with complex-step.
+
+    Parameters
+    ----------
+    y : float or complex
+        The length of the side opposite the angle being determined.
+    x : float or complex
+        The length of the side adjacent to the angle being determined.
+
+    Returns
+    -------
+    ndarray
+        The angle whose opposite side has length y and whose adjacent side has length x.
+    """
+    a = np.real(y)
```

Comment:
a, b, c, d temporaries could be moved under the if block and just use y and x under the else block.